[mono][mini] Fix conversion from r4/r8 to native u

BrzVlad opened the pull request:
When converting from float point to unsigned integer, we always do a conversion to u8, then do a simple integer conversion with range checks. The problem is that the first conversion to u8 pushes a STACK_I8 and doing a further conversion to u8 is not actually a nop since it will throw if the result of the previous conversion is negative.

Fixes https://github.com/dotnet/runtime/issues/69794

BrzVlad committed:
[mono][mini] Fix conversion from r4/r8 to native u

When converting from float point to unsigned integer, we always do a conversion to u8, then do a simple integer conversion with range checks. The problem is that the first conversion to u8 pushes a STACK_I8 and doing a further conversion to u8 is not actually a nop since it will throw if the result of the previous conversion is negative.

BrzVlad committed:
Enable tests

vargaz commented on src/mono/mono/mini/method-to-ir.c:
```diff
@@ -8992,7 +8995,10 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				/* floats are always signed, _UN has no effect */
 				ADD_UNOP (CEE_CONV_OVF_U8);
-				ADD_UNOP (il_op);
+				if (TARGET_SIZEOF_VOID_P == 8 && il_op == MONO_CEE_CONV_OVF_U)
```
This could be if (!(....)) ADD_UNOP..

SamMonoRT commented:
@BrzVlad - are CI failures related ?

BrzVlad commented:
Yes but only happening on arm64 apparently. VM software would be useful here

BrzVlad committed:
[mono][jit] Reuse conv ovf methods used by interpreter

These follow CoreCLR implementation and don't rely on undefined behavior from out of range conversions.
